src/main/java/com/purbon/kafka/topology/ExecutionPlan.java

```diff
@@ -81,9 +82,12 @@ public void run(boolean dryRun) throws IOException {
   }
 
   private void execute(Action action, boolean dryRun) throws IOException {
-    LOGGER.debug(String.format("Execution action %s (dryRun=%s)", action, dryRun));
+    LOGGER.debug("Execution action {} (dryRun={}})", action, dryRun);
     if (dryRun) {
-      outputStream.println(action);
+      String actions = action.toString();
+      if (StringUtils.isNotBlank(actions)) {
```

Comment:
it does the same as before but just not print something in case the action.toString() is empty.

outputStream.println(action); is the same as outputStream.println(action.toString())

Reply:
Sure, 
 I guess here I am thinking.

* what does it mean for action.toString to be empty? can it be a check we can add to the action class?

Reply:
yes, but looks like it's not needed anymore as i also changed the BaseAction already :-)